packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,229 @@
+                    `editing and saving temporary file: ${filePath}`));
+                if (input === null) {
+                    throw new ImperativeError({
+                        msg: TextUtils.chalk.red(`No input provided. Command terminated. Stashed file will persist: ${filePath}`)
+                    });
+                }else{
+                    return true;
+                }
+            case Prompt.continueToUpload:
+                input = await CliUtils.readPrompt(TextUtils.chalk.green(`The version of the document you were editing has changed.` +
+                    `Continue uploading current edits? Y/n\n${filePath}\n`));
+                if (input === null) {
+                    throw new ImperativeError({
+                        msg: TextUtils.chalk.red(`No input provided. Command terminated. Stashed file will persist: ${filePath}`)
+                    });
+                }
+                if (input.toLowerCase() == 'y'){
+                    //upload
+                    return true;
+                } else {
+                    //open diff, keep editing
+                    return false;
+                }
+        }
+    }
+
+    public static async fileComparison(session: AbstractSession, commandParameters: IHandlerParameters): Promise<IZosFilesResponse>{
+        const handlerDs = new LocalfileDatasetHandler;
+        const handlerUss = new LocalfileUssHandler;
+        const helper = new CompareBaseHelper(commandParameters);
+        const gui = ProcessUtils.isGuiAvailable();
+        const options: IDiffOptions = {
+            name1: "local file",
+            name2: "mainframe file"
+        };
+
+        if(gui === GuiResult.GUI_AVAILABLE){
+            helper.browserView = true;
+        }
+
+        const lf = await handlerDs.getFile1(session, commandParameters.arguments, helper);
+        let mfds: string | Buffer;
+        if (commandParameters.positionals.includes('ds')){
+            mfds = await handlerDs.getFile2(session, commandParameters.arguments, helper);
+        }else{
+            mfds = await handlerUss.getFile2(session, commandParameters.arguments, helper);
+        }
+        // Editor will open with local file if default editor was set
+        return await helper.getResponse(helper.prepareContent(lf), helper.prepareContent(mfds), options);
+    }
+
+    public static async makeEdits(commandParameters: IHandlerParameters): Promise<void>{
+        const tmpDir = commandParameters.arguments.localFilePath;
+        // Perform file comparison: show diff in terminal, open lf in editor
+        if (commandParameters.arguments.editor){
+            await ProcessUtils.openInEditor(tmpDir, commandParameters.arguments.editor);
+        }
+        await this.promptUser(Prompt.doneEditing, tmpDir);
+
+    }
+
+    public static async uploadEdits(session: AbstractSession, commandParameters: IHandlerParameters, lfFile: LocalFile): Promise<boolean>{
```

Comment:
I know that you asked for an early review focused on the programming logic, but I will make this comment anyway:

I really like your line comments before individual paragraphs of logic. I think that you should additionally place typedoc-style block comments before every function.

Reply:
Oh yes! Thank you and sorry I forgot to add those in. Will fix!